main/src/main/scala/sbt/plugins/DependencyReportPlugin.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * sbt
+ * Copyright 2023, Scala center
+ * Copyright 2011 - 2022, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * Licensed under Apache License 2.0 (see LICENSE)
+ */
+
+package sbt
+package plugins
+
+import sbt.PluginTrigger.AllRequirements
+import sbt.Project._
+import sbt.librarymanagement.Configurations.{ Compile, Test }
+
+object DependencyReportPlugin extends AutoPlugin {
```

Comment:
For sbt 2.x, I wonder if we could consolidate 3 dependency-tree plugins into one. When we in-sourced it to sbt/sbt we had to split it into two because of the amount of setting keys it introduced to all subproject slowed down large builds. If we can reduce the number of injected settings we use this opportunity to do so.

Reply:
My concern is that there's some settings that are currently unsupported as they are ugly to support via command line argument (e.g. Graph customization options as part of full dependency tree plugin).

I don't know whether people actually use these settings though.

Reply:
@mkurz What is your opinion on Eugene's idea of consolidating 3 dependency-tree plugins into one?

Reply:
> I can confirm that the lack of consolidation is rather mystifying for users. It comes up from time to time on Discord.

@SethTisue I guess indeed better to go for full consolidation then. We will lose some flexibility (e.g. inability to customize graph appearance detail), but we will end up with a simplified implementation, vastly less settings key and less user confusion.

I will wait for a few days for further discussion, then proceed with full consolidation as per Eugene's comment.